internal/logger/targets.go

```diff
@@ -108,3 +111,37 @@ func AddTarget(t Target) error {
 
 	return nil
 }
+
+func cancelAllTargets() {
+	for _, tgt := range Targets() {
+		tgt.Cancel()
+	}
+}
+
+func initTargets(cfg Config) (tgts []Target, err error) {
+	for _, l := range cfg.HTTP {
+		if l.Enabled {
+			t := http.New(l)
+			if err = t.Init(); err != nil {
+				return tgts, err
+			}
+			tgts = append(tgts, t)
+		}
+	}
+	return tgts, err
+}
+
+// UpdateTargets swaps targets with newly loaded ones from the cfg
+func UpdateTargets(cfg Config) error {
+	cancelAllTargets() // cancel running targets
+	updated, err := initTargets(cfg)
+	if err != nil {
+		return err
+	}
+
+	swapMu.Lock()
+	targets = updated
+	atomic.StoreInt32(&nTargets, int32(len(updated)))
+	swapMu.Unlock()
```

Comment:
```suggestion
	swapMu.Lock()
	atomic.StoreInt32(&nTargets, int32(len(updated)))
	cancelAllTargets(targets)
	targets = updated
	swapMu.Unlock()
```

Change `cancelAllTargets` to:
```Go
func cancelAllTargets(targets []Target) {
	for _, tgt := range targets {
		tgt.Cancel()
	}
}
```
AFAICT this should ensure that we have consistent logging, meaning we don't drop and and we shouldn't have duplicates.

Reply:
Done